### PR TITLE
fix: call notation-go libs instead of using notation binary

### PR DESCRIPTION
**Pull request by Andreea-Lupu**
Signed-off-by: Roxana Nemulescu <roxana.nemulescu@gmail.com>

fix: add loading notation path

Signed-off-by: Andreea-Lupu <andreealupu1470@yahoo.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -105,7 +105,7 @@ check-skopeo:
 
 $(NOTATION):
 	mkdir -p $(TOOLSDIR)/bin
-	curl -Lo notation.tar.gz https://github.com/notaryproject/notation/releases/download/v0.7.1-alpha.1/notation_0.7.1-alpha.1_linux_amd64.tar.gz
+	curl -Lo notation.tar.gz https://github.com/notaryproject/notation/releases/download/v1.0.0-rc.1/notation_1.0.0-rc.1_linux_amd64.tar.gz
```
Comment (rchincha) [inline Makefile:108]: can we make this a variable? instead of hardcoding
NOTATION_VERSION=1.0.0-rc.1

File: go.mod [hunk 2]
```diff
@@ -55,10 +55,11 @@ require (
 	github.com/aws/aws-sdk-go-v2/service/dynamodb v1.18.0
 	github.com/containers/image/v5 v5.23.0
 	github.com/gobwas/glob v0.2.3
-	github.com/notaryproject/notation-go v0.12.0-beta.1
+	github.com/notaryproject/notation-go v1.0.0-rc.1
```
Comment (rchincha) [inline go.mod:58]: 👍

File: pkg/api/controller_test.go [hunk 3]
```diff
@@ -4023,49 +3994,45 @@ func TestImageSignatures(t *testing.T) {
 			Convey("Validate corrupted signature", func() {
 				// verify with corrupted signature
 				resp, err = resty.R().SetQueryParam("artifactType", notreg.ArtifactTypeNotation).Get(
-					fmt.Sprintf("%s/oras/artifacts/v1/%s/manifests/%s/referrers", baseURL, repoName, digest.String()))
+					fmt.Sprintf("%s/v2/%s/referrers/%s", baseURL, repoName, digest.String()))
```
Comment (rchincha) [inline pkg/api/controller_test.go:3997]: so we are moving to oci references here instead of oras one?

Comment (rchincha) [reply pkg/api/controller_test.go:3997]: which is fine, we should eventually deprecate oras completely.

**Comment by rchincha** [comment]
@Andreea-Lupu some minor comments, but looks good to merge, pls rebase once

**Comment by peusebiu** [comment]
sync LGTM

File: pkg/api/controller_test.go (hunk 3 above)
Comment (rchincha) [reply pkg/api/controller_test.go:3997]: But let's do this after conferring with oras team.

**Review by andaaron — COMMENTED** [review]
Looks good overall, but I do have a concern which I mentioned in the other comments.

**Review by andaaron — DISMISSED** [review]

File: pkg/api/routes.go [hunk 4]
```diff
@@ -440,7 +440,7 @@ func getReferrers(ctx context.Context, routeHandler *RouteHandler,
 			errSync := ext.SyncOneImage(ctx, routeHandler.c.Config, routeHandler.c.StoreController,
 				name, digest.String(), sync.OCIReference, routeHandler.c.Log)
 			if errSync != nil {
-				routeHandler.c.Log.Error().Err(err).Str("name", name).Str("digest", digest.String()).Msg("unable to get references")
+				routeHandler.c.Log.Error().Err(err).Str("name", name).Str("digest", digest.String()).Msg("unable to get references") //nolint:lll
```
Comment (rchincha) [inline pkg/api/routes.go:443]: No need for the nolint, just break it into newlines before/after the dot

File: pkg/api/controller_test.go [hunk 5]
```diff
@@ -3961,54 +3960,29 @@ func TestImageSignatures(t *testing.T) {
...
-			cmd = exec.Command("notation", "cert", "generate-test", "--trust", "bad")
-			err = cmd.Run()
+			err = test.GenerateNotationCerts(tdir, "bad")
 			So(err, ShouldBeNil)
 
-			// sign the image
 			image := fmt.Sprintf("localhost:%s/%s:%s", port, repoName, "1.0")
-			cmd = exec.Command("notation", "sign", "--key", "good", "--plain-http", image)
-			err = cmd.Run()
+			err = test.SignUsingNotation("good", image, tdir)
```
Comment (rchincha) [inline pkg/api/controller_test.go:3975]: Call this SignWithNotation()

File: pkg/api/controller_test.go [hunk 6]
```diff
@@ -3961,54 +3960,29 @@ func TestImageSignatures(t *testing.T) {
...
 			So(err, ShouldBeNil)
 
-			// verify the image
-			cmd = exec.Command("notation", "verify", "--cert", "good", "--plain-http", image)
-			out, err := cmd.CombinedOutput()
+			err = test.VerifyNotarySignature(image, tdir)
```
Comment (rchincha) [inline pkg/api/controller_test.go:3978]: Call this VerifyWithNotation()

File: pkg/extensions/sync/on_demand.go [hunk 7]
```diff
@@ -300,14 +300,14 @@ func syncRun(regCfg RegistryConfig,
 			Err(err).Msgf("couldn't get upstream image %s cosign manifest", upstreamImageRef.DockerReference())
 	}
 
-	refs, err := sig.getNotaryRefs(upstreamRepo, upstreamImageDigest.String())
+	index, err := sig.getOCIRefs(upstreamRepo, upstreamImageDigest.String())
 	if err != nil {
 		log.Error().Str("errorType", common.TypeOf(err)).
-			Err(err).Msgf("couldn't get upstream image %s notary references", upstreamImageRef.DockerReference())
+			Err(err).Msgf("couldn't get upstream image %s oci references", upstreamImageRef.DockerReference())
```
Comment (rchincha) [inline pkg/extensions/sync/on_demand.go:306]: s/oci/OCI/g

**Comment by rchincha** [comment]
@peusebiu can you pls review the sync part of this PR?

**Comment by peusebiu** [comment]
sync looks good, I also see a few bug fixes, thank you Andreea!

File: pkg/extensions/sync/signatures.go [hunk 8]
```diff
@@ -608,3 +608,15 @@ func getCosignTagFromImageDigest(digestStr string) string {
 
 	return digestStr
 }
+
+func getNotationManifestsFromOCIRefs(ociRefs ispec.Index) []ispec.Descriptor {
+	notaryManifests := []ispec.Descriptor{}
+
+	for _, ref := range ociRefs.Manifests {
+		if ref.ArtifactType == notreg.ArtifactTypeNotation {
+			notaryManifests = append(notaryManifests, ref)
```
Comment (peusebiu) [inline pkg/extensions/sync/signatures.go:617]: hmm, why is this not covered?

should be covered from all the notation tests, right?

Comment (peusebiu) [reply pkg/extensions/sync/signatures.go:617]: nvm, it's ok, is not covered because of the order of signing in tests (first cosign then notation)

**Review by andaaron — APPROVED** [review]

**Comment by rchincha** [comment]
Pls resolve conflicts and push again

**Review by andaaron — APPROVED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**